ecodse_funtime_alpha/data.py

```diff
@@ -3,23 +3,61 @@
 
 import tensorflow as tf
 
-import matplotlib.pyplot as plt
 from sklearn.preprocessing import MultiLabelBinarizer
 
 
 def preprocess_image(image):
+    """
+    Decode a jpeg and normalize it to [0, 1] range.
+
+    Parameters
+    ----------
+    image : tf tensor
+        Raw image as a tf tensor
+
+    Returns
+    -------
+        tf tensor in float32
+    """
     image = tf.io.decode_jpeg(image, dct_method="INTEGER_ACCURATE")
     image = tf.cast(image, tf.float32)
     image /= 255.0  # normalize to [0,1] range
     return image
 
 
 def load_and_preprocess_image(path):
+    """
+    Read a jpeg and normalize it to [0, 1] range.
+
+    Parameters
+    ----------
+    path : str
+        Path to the jpeg to read
+
+    Returns
+    -------
+        tf tensor in float32
+    """
     image = tf.io.read_file(path)
     return preprocess_image(image)
 
 
 def get_dataset(image_dir, labels_csv):
+    """
+    Prepare the dataset for training. Images are normalized to [0, 1] range. Labels are n-hot encoded.
+
+    Parameters
+    ----------
+    image_dir : str
+        Path to the directory containing the images
+    labels_csv : str
+        Path to the csv file
+
+    Returns
+    -------
+        tf.data.ZipDataset
+            Elements of the dataset are structured (image, labels)
+    """
     samples = []
     with open(labels_csv) as csvfile:
         reader = csv.reader(csvfile, delimiter=',')
```

Comment:
The way to skip the header is `reader.next()`, then enumerate and `if i` becomes useless

Reply:
for `'{}.jpg'.format(row[0])` use fstring instead `f'{row[0]}.jpg'`

Reply:
Done